### PR TITLE
Add macro to match the adapter

### PR DESCRIPTION
**Pull request by dati91**
Refactor WebBluetooth Adapter code.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because there are no webbluetooth tests yet.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11503)

### CODE REVIEW

**Comment by dati91** [comment]
It seems to me that the error does not related to this patch.

**Comment by jdm** [comment]
The error is #11494.

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -62,6 +62,15 @@ macro_rules! return_if_cached(
     );
 );
 
+macro_rules! get_adapter_or_return_error(
```
Comment (KiChjang) [inline components/net/bluetooth_thread.rs]: Can this instead be a function? I don't see why it should be a macro.

Comment (jdm) [inline components/net/bluetooth_thread.rs]: Because it needs to return from the caller?

**Pull request merged**